org/postgresql/core/Parser.java

```diff
@@ -90,11 +90,16 @@
                     i++; // make sure the coming ? is not treated as a bind
                 } else
                 {
-                    if (bindPositions == null)
-                        bindPositions = new ArrayList<Integer>();
-                    bindPositions.add(nativeSql.length());
-                    int bindIndex = bindPositions.size();
-                    nativeSql.append(NativeQuery.bindName(bindIndex));
+                    if (withParameters)
+                    {
+                        if (bindPositions == null)
+                            bindPositions = new ArrayList<Integer>();
+                        bindPositions.add(nativeSql.length());
+                        int bindIndex = bindPositions.size();
+                        nativeSql.append(NativeQuery.bindName(bindIndex));
+                    }
+                    else
+                        nativeSql.append('?');
```

Comment:
Please put "shortest branch" first, please always use braces (it seems to be the project style)